### PR TITLE
use happy path for better code readability

### PR DESCRIPTION
**Pull request by mschneider82**
hi, it was hard to follow the "good" code, i refactored/cleaned up many segments files. Also there was something like:

```
  var s string
  if s == "" { ... }
```
which is always true because the string was defined in the line before.
I also changed some if .. else .. things to switch statements with default. 

i dont think that i breaked something, but please review. But dont wait too long since its quite hard to keep this change up to date , since it touches many files. Thanks

### CODE REVIEW

**Comment by justjanne** [comment]
This looks like a great PR, sadly it’d need significant rebasing to get it merged. If you could do that, I’d gladly merge it for v1.21.0

**Comment by mschneider82** [comment]
done please review @justjanne
